test/app/components/custom_test_controller_component.rb

```diff
@@ -0,0 +1,9 @@
+# frozen_string_literal: true
+
+class CustomTestControllerComponent < ViewComponent::Base
+  self.test_controller = "CustomTestControllerController"
```

Comment:
A lot of rails apis would wrap class_attributes to provide a method so you can do `test_controller("CustomTestControllerController")` or the like. 

Also is it necessary for this to be a string vs a class? I believe we want a string version as something that can get set in application.rb or initializers without having to worry about the class being autoloaded. However, I wonder at this point if we can't just use classes.

Reply:
Personally, I like the less DSL-y `self.test_controller=` interface over the more magical `test_controller 'CustomTestControllerController'`, but regardless, I tried not to change the interface here to leave things backwards compatible and not introduce more changes that I needed to!